htlcswitch/hop/iterator.go

```diff
@@ -113,6 +120,98 @@ func (r *sphinxHopIterator) ExtractErrorEncrypter(
 	return extracter(r.ogPacket.EphemeralKey)
 }
 
+// BlindingProcessor is an interface that provides the cryptographic operations
+// required for processing blinded hops.
+type BlindingProcessor interface {
+	// DecryptBlindedHopData decrypts a blinded blob of data using the
+	// ephemeral key provided.
+	DecryptBlindedHopData(ephemPub *btcec.PublicKey,
+		encryptedData []byte) ([]byte, error)
+}
+
+// BlindingKit contains the components required to extract forwarding
+// information for hops in a blinded route.
+type BlindingKit struct {
+	// BlindingPoint holds a blinding point that was passed to the node via
+	// update_add_htlc's TLVs.
+	BlindingPoint *btcec.PublicKey
+
+	// ForwardingInfo uses the ephemeral blinding key provided to decrypt
+	// a blob of encrypted data provided in the onion and obtain the
+	// forwarding information for the blinded hop.
+	ForwardingInfo func(*btcec.PublicKey, []byte) (*ForwardingInfo,
+		error)
+
+	// DisableBlindedFwd provides the ability to optionally disable blinded
+	// route forwarding. Payloads will simply be rejected as invalid if
+	// they contain blinding points, as our node does not want to forward
+	// them.
+	DisableBlindedFwd bool
+}
+
+// MakeBlindingKit produces a kit that is used to decrypt and decode
+// forwarding information for hops in blinded routes.
+func MakeBlindingKit(processor BlindingProcessor,
+	blindingPoint *btcec.PublicKey, incomingAmount lnwire.MilliSatoshi,
+	incomingCltv uint32, disableBlinding bool) *BlindingKit {
+
+	return &BlindingKit{
+		BlindingPoint: blindingPoint,
+		ForwardingInfo: func(blinding *btcec.PublicKey,
+			data []byte) (*ForwardingInfo, error) {
+
+			decrypted, err := processor.DecryptBlindedHopData(
```

Comment:
See comment above re the processor being extended to cover these duties. Then we just return the _interface_ (or rather accept it elsewhere) that's capable of the decrypt+decode.